### DISCUSSION TITLE
Arduino ESP32 cloning has exit with 128

### DISCUSSION THREAD

**Discussion by M-Rizwan-Anwar** [espressif/arduino-esp32#9290] [Q&A] [answered]
I am getting this error when adding arduino as component. can anyone help me?

**Comment by SuGlider** [comment]
@M-Rizwan-Anwar - please tell us how you have set up the Arduino as Component environment. We need Arduino Core version, IDF version, folder structure and any other information that could help.
If possible, create a repository with your basic project example and post it here.

**Comment by SuGlider** [comment]
@M-Rizwan-Anwar - You can try this folder/setup example: https://github.com/espressif/arduino-esp32/tree/master/idf_component_examples/Hello_world

It uses the Arduino 2.0.14 with IDF 4.4.6. The example is configured for ESP32.
Make sure that your IDF is in this version. Run `install.sh` and `export.sh` as usual, then build it using `idf.py build`.

**Comment by M-Rizwan-Anwar** [comment]
I tried it with IDF 5.2, I created an example project and then from Pallete I tried to add Arduino as an component but it said "ESP32 Cloning has exit with 128" then I manually added "Component folder in my project folder from Github. Then when I tried to build it showed Implicit function error.

**Reply by me-no-dev** [reply]
> I tried it with IDF 5.2

This repository master supports only ESP-IDF v5.1.x

**Comment by M-Rizwan-Anwar** [comment, marked as answer]
Thanks. It worked for me.

**Comment by byNMI** [comment]
when we can add esp32 as idf component to v5.3.x ?

**Reply by me-no-dev** [reply]
you can give it a shot now with `release/v3.1.x` branch
